src/Resource/DependencyExtractor.php

```
@@ -0,0 +1,212 @@
+            $composer->getPackage()->getMinimumStability(),
+            $composer->getPackage()->getStabilityFlags(),
+        );
+        $repositorySet->addRepository(
+            new Repository\CompositeRepository($composer->getRepositoryManager()->getRepositories()),
+        );
+        $versionSelector = new Package\Version\VersionSelector(
+            $repositorySet,
+            new Repository\PlatformRepository([], $composer->getConfig()->get('platform')),
+        );
+
+        foreach ($packages as $packageName => $package) {
```

PHPMD (2.15.0)

Avoid unused local variables such as '$package'. (undefined)

(UnusedLocalVariable)